eng/pipelines/checkout-unix-task.yml

```diff
@@ -0,0 +1,15 @@
+# Shallow checkout sources on Unix
+
+steps:
+  - checkout: none
+
+  - script: |
+      @echo on
+      git init
+      git config --local checkout.workers 0
+      git config --local fetch.parallel 0
+      git remote add origin "$(Build.Repository.Uri)"
+      git fetch --progress --no-tags --no-auto-maintenance --depth=1 origin "$(Build.SourceVersion)"
```

Comment:
is there a specific reason to include `--progress`? seems to clutter the CI logs